### PR TITLE
Dashboard SSO integration

### PR DESCRIPTION
**Pull request by VictiniX888**
### Description

(_What have been done/changed_)

- Protected dashboard with SSO
- Dynamically display users' name/email on dashboard by pulling information from database (closes HIB-111)
- Implemented logout button on dashboard (closes HIB-110)

### Testing plan

(_How will we test the changes requested on this PR?_)

- [ ] ...

### CODE REVIEW

File: apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts [hunk 1]
```diff
@@ -14,11 +17,44 @@ export interface UseHibiscusUser {
...
-    firstName: 'Vincent',
-  });
+  const [user, setUser] = useState<HibiscusUser>(null);
+
+  async function getUserProfile() {
+    // Get user profile from db
+    const supabase = container.resolve(HibiscusSupabaseClient);
+    const profile = await supabase.getUserProfile(
+      getCookie(process.env.NEXT_PUBLIC_HIBISCUS_ACCESS_COOKIE_NAME) as string,
+      getCookie(process.env.NEXT_PUBLIC_HIBISCUS_REFRESH_COOKIE_NAME) as string
```
Comment (snokpok) [inline apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts:27]: nit: try using and adding stuff from `@hibiscus/env`

File: apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts [hunk 2]
```diff
@@ -14,11 +17,44 @@ export interface UseHibiscusUser {
...
+        firstName: profile.first_name,
+      });
+    } else {
+      // Set user's name and tag to be their email as temporary placeholder
+      // Assume we only show dashboard when user is logged in
+      const user = await supabase
+        .getClient()
+        .auth.getUser(
+          getCookie(
+            process.env.NEXT_PUBLIC_HIBISCUS_ACCESS_COOKIE_NAME
```
Comment (snokpok) [inline apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts:43]: like above

File: apps/dashboard/layouts/portal-layout.tsx [hunk 3]
```diff
@@ -8,6 +8,9 @@ export type PortalLayoutProps = React.PropsWithChildren;
 
 function PortalLayout({ children }: PortalLayoutProps) {
   const { user } = useHibiscusUser();
+  if (user == null) {
+    return <></>;
```
Comment (snokpok) [inline apps/dashboard/layouts/portal-layout.tsx:12]: maybe redirect back to SSO instead?

File: apps/dashboard/middleware.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { middlewareHandler } from '@hibiscus/sso-client';
+
+export const middleware = middlewareHandler(
+  `${process.env.NEXT_PUBLIC_PORTAL_URL}/api/callback`
```
Comment (snokpok) [inline apps/dashboard/middleware.ts:4]: nit: use `@hibiscus/env`

File: apps/dashboard/pages/api/callback.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { NextApiHandler } from 'next';
+import { callbackApiHandler } from '@hibiscus/sso-client';
+
+const handler: NextApiHandler = callbackApiHandler(
+  process.env.NEXT_PUBLIC_PORTAL_URL
```
Comment (snokpok) [inline apps/dashboard/pages/api/callback.ts:5]: like above

File: apps/dashboard/pages/index.tsx [hunk 6]
```diff
@@ -5,10 +5,15 @@ import { Search } from '@hibiscus/ui-kit-2023';
...
 import useHibiscusUser from '../hooks/use-hibiscus-user/use-hibiscus-user';
 import { getColorsForRole } from '../common/role.utils';
+import { HibiscusRole } from '@hibiscus/types';
 
 export function Index() {
   const { user } = useHibiscusUser();
-  const userColors = getColorsForRole(user.role);
+  if (user == null) {
+    return <></>;
+  }
```
Comment (snokpok) [inline apps/dashboard/pages/index.tsx:14]: like above

**Review by snokpok — CHANGES_REQUESTED** [review]
small fixes

File: apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts (hunk 1 above)
Comment (VictiniX888) [reply apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts]: Done!

File: apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts (hunk 2 above)
Comment (VictiniX888) [reply apps/dashboard/hooks/use-hibiscus-user/use-hibiscus-user.ts]: Done!

File: apps/dashboard/middleware.ts (hunk 4 above)
Comment (VictiniX888) [reply apps/dashboard/middleware.ts]: Done!

File: apps/dashboard/pages/api/callback.ts (hunk 5 above)
Comment (VictiniX888) [reply apps/dashboard/pages/api/callback.ts]: Done!

**Review by snokpok — APPROVED** [review]
lgtm!

**Pull request merged**